elements/lisk-chain/src/data_access/storage.ts

```diff
@@ -12,291 +12,537 @@
+
+			return value;
+		} catch (error) {
+			if (error instanceof NotFoundError) {
+				return undefined;
+			}
+			throw error;
+		}
 	}
 
 	/*
 		ConsensusState
 	*/
 	public async getConsensusState(key: string): Promise<string | undefined> {
-		const value = await this._storage.entities.ConsensusState.getKey(key);
-
-		return value;
+		try {
+			const value = await this._db.get<string>(
+				`${DB_KEY_CONSENSUS_STATE}${key}`,
+			);
+
+			return value;
+		} catch (error) {
+			if (error instanceof NotFoundError) {
+				return undefined;
+			}
+			throw error;
+		}
 	}
 
 	/*
 		Accounts
 	*/
-	public async getAccountsByPublicKey(
-		arrayOfPublicKeys: ReadonlyArray<string>,
-	): Promise<AccountJSON[]> {
-		const accounts = await this._storage.entities.Account.get(
-			// eslint-disable-next-line camelcase
-			{ publicKey_in: arrayOfPublicKeys },
-			{ limit: arrayOfPublicKeys.length },
+	public async getAccountByAddress(address: string): Promise<AccountJSON> {
+		const account = await this._db.get<AccountJSON>(
+			`${DB_KEY_ACCOUNTS_ADDRESS}${address}`,
 		);
 
-		return accounts;
+		return account;
 	}
 
-	public async getAccountByAddress(address: string): Promise<AccountJSON> {
-		const account = await this._storage.entities.Account.getOne(
-			{ address },
-			{ limit: 1 },
-		);
+	public async getAccountsByPublicKey(
+		arrayOfPublicKeys: ReadonlyArray<string>,
+	): Promise<AccountJSON[]> {
+		const addresses = arrayOfPublicKeys.map(getAddressFromPublicKey);
 
-		return account;
+		return this.getAccountsByAddress(addresses);
 	}
 
 	public async getAccountsByAddress(
 		arrayOfAddresses: ReadonlyArray<string>,
 	): Promise<AccountJSON[]> {
-		const accounts = await this._storage.entities.Account.get(
-			// eslint-disable-next-line camelcase
-			{ address_in: arrayOfAddresses },
-			{ limit: arrayOfAddresses.length },
-		);
+		const accounts = [];
+		for (const address of arrayOfAddresses) {
+			const account = await this.getAccountByAddress(address);
+			accounts.push(account);
+		}
 
 		return accounts;
 	}
 
+	// TODO: Remove this with issue #5259
 	public async getDelegates(): Promise<AccountJSON[]> {
-		const accounts = await this._storage.entities.Account.get(
-			{ isDelegate: true },
-			// Sort address:asc is always added in the storage
-			{ limit: null, sort: ['totalVotesReceived:desc'] },
-		);
+		const stream = this._db.createReadStream({
+			gte: getFirstPrefix(DB_KEY_ACCOUNTS_ADDRESS),
+			lte: getLastPrefix(DB_KEY_ACCOUNTS_ADDRESS),
+		});
+		const accounts = await new Promise<AccountJSON[]>((resolve, reject) => {
+			const accountJSONs: AccountJSON[] = [];
+			stream
+				.on('data', ({ value }) => {
+					const { username } = value as AccountJSON;
+					if (username) {
+						accountJSONs.push(value);
+					}
+				})
+				.on('error', error => {
+					reject(error);
+				})
+				.on('end', () => {
+					resolve(accountJSONs);
+				});
+		});
+		accounts.sort((a, b) => {
```

Comment:
maybe there're more places where we need to sort BigInts so this could be better in a helper function?

Reply:
I think this sorting is also done in the `dpos`, and as mentioned in this function comment, this will be removed.
Also, it's hard to generalize this `sort`, because this is very specific to the delegate sorting, and this should be only done in the `dpos` library